subsys/testsuite/ztest/src/ztest_posix.c

```diff
@@ -161,10 +162,18 @@ static bool z_ztest_testargs_contains(const char *suite_name, const char *test_n
 		suite_arg = strtok_r(suite_test_pair, ":", &last_arg);
 		test_arg = strtok_r(NULL, ":", &last_arg);
 
+		/* Validate format: must be test_suite::* or test_suite::test_case */
+		if (!suite_arg || !test_arg || strtok_r(NULL, ":", &last_arg) != NULL) {
+			/* Invalid format, report error and exit */
+			suite_test_pair = strtok_r(NULL, ",", &last_suite_test_pair);
+			nsi_print_error_and_exit(
+				"Invalid test argument format: '%s'. Expected 'suite::test' or 'suite::*'.\n",
+				suite_test_pair);
```

Comment:
Aren't you jumping to the next `suite_test_pair` here?
Meaning, shouldn't this just be:
```suggestion
			nsi_print_error_and_exit(
				"Invalid test argument format: '%s'. Expected 'suite::test' or 'suite::*'.\n",
				suite_test_pair);
```
or even
```suggestion
			nsi_print_error_and_exit(
				"Invalid test argument format '%s'. Expected a set of pairs like"
				"'suite::test' or 'suite::*', got instead '%s'.\n",
				test_args, suite_test_pair);
```